### PR TITLE
Adding createFlowUnionType in place of createUnionTypeAnnotation without breaking change

### PR DESCRIPTION
**Pull request by Beraliv**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes https://github.com/Berkmann18/babel/pull/271
| Patch: Bug Fix?          | No
| Major: Breaking Change?  | No
| Minor: New Feature?      | Yes (technically)
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  | No
| License                  | MIT

Adding new Flow function `createFlowUnionType` which is more descriptive than `createUnionTypeAnnotation`

However, I save `createUnionTypeAnnotation` for backward compatibility (it can be deleted in any major release, I can add TODO if needed)

Connected to changes in https://github.com/babel/babel/pull/11378

### CODE REVIEW

File: packages/babel-traverse/src/path/inference/inferer-reference.js [hunk 1]
```diff
@@ -99,6 +99,10 @@ function getTypeAnnotationBindingConstantViolations(binding, path, name) {
     return t.createTSUnionType(types);
   }
 
+  if (t.createFlowUnionType) {
```
Comment (Beraliv) [inline packages/babel-traverse/src/path/inference/inferer-reference.js:102]: Check if it's available (in new versions only)
Old version will go further

The same for other cases below

File: packages/babel-types/scripts/generators/flow.js [hunk 2]
```diff
@@ -128,6 +128,8 @@ lines.push(
   `declare function createTypeAnnotationBasedOnTypeof(type: 'string' | 'number' | 'undefined' | 'boolean' | 'function' | 'object' | 'symbol'): ${NODE_PREFIX}TypeAnnotation`,
   // eslint-disable-next-line max-len
   `declare function createUnionTypeAnnotation(types: Array<${NODE_PREFIX}FlowType>): ${NODE_PREFIX}UnionTypeAnnotation`,
+  // eslint-disable-next-line max-len
+  `declare function createFlowUnionType(types: Array<${NODE_PREFIX}FlowType>): ${NODE_PREFIX}UnionTypeAnnotation`,
```
Comment (Beraliv) [inline packages/babel-types/scripts/generators/flow.js:132]: not removing old declaration, adding the new one

File: packages/babel-types/src/builders/flow/createUnionTypeAnnotation.js [hunk 3]
```diff
@@ -1,19 +1,15 @@
...
 ): Object {
-  const flattened = removeTypeDuplicates(types);
-
-  if (flattened.length === 1) {
-    return flattened[0];
-  } else {
-    return unionTypeAnnotation(flattened);
-  }
+  return createFlowUnionType(types);
 }
```
Comment (nicolo-ribaudo) [inline packages/babel-types/src/builders/flow/createUnionTypeAnnotation.js:15]: Nit:

```js
export { createFlowUnionType as createUnionTypeAnnotation };
```

Comment (Beraliv) [reply packages/babel-types/src/builders/flow/createUnionTypeAnnotation.js:15]: Yeah, this is rational

Comment (Beraliv) [reply packages/babel-types/src/builders/flow/createUnionTypeAnnotation.js]: ✅ Completed

**Comment by Beraliv** [comment]
@nicolo-ribaudo is something wrong with test coverage and other tests or did I break anything?

**Comment by nicolo-ribaudo** [comment]
Yeah, it's the codecov integration which is buggy.

**Comment by nicolo-ribaudo** [comment]
@Beraliv Would you like to open a PR to the `next-8-dev` (Babel 8) branch removing the checks needed for compatibility with Babel 7?

I'm merging `master` in `next-8-dev` right now.

**Comment by Beraliv** [comment]
> @Beraliv Would you like to open a PR to the `next-8-dev` (Babel 8) branch removing the checks needed for compatibility with Babel 7?
> 
> I'm merging `master` in `next-8-dev` right now.

Sure, will be ready this night!

**Pull request merged**